scipy/optimize/_shgo.py

```diff
@@ -48,14 +48,14 @@ def shgo(func, bounds, args=(), constraints=None, n=100, iters=1, callback=None,
 
         Each constraint is defined in a dictionary with fields:
 
-            type : str
-                Constraint type: 'eq' for equality, 'ineq' for inequality.
-            fun : callable
-                The function defining the constraint.
-            jac : callable, optional
-                The Jacobian of `fun` (only for SLSQP).
-            args : sequence, optional
-                Extra arguments to be passed to the function and Jacobian.
+        type : str
```

Comment:
really wish we could use `NonlinearConstraint` as default and dump the dicts. Anyway, `trust-constr` accepts new-style constraints so is statement in the `Note` box correct? https://22669-1460385-gh.circle-artifacts.com/0/html-scipyorg/generated/scipy.optimize.shgo.html#scipy.optimize.shgo

Reply:
It just wasn't updated when trust-constr was added. I snuck it in here for you.

Reply:
Fixed note box:
![image](https://user-images.githubusercontent.com/6570539/102744789-71ef2300-430f-11eb-98ee-1907a0df6d20.png)

https://23582-1460385-gh.circle-artifacts.com/0/html-scipyorg/generated/scipy.optimize.shgo.html#scipy.optimize.shgo